### ISSUE TITLE
Modify lambda to also accept S3 events via SNS

### ISSUE THREAD

**Issue by hwine** [mozilla-services/fx-sig-verify#28]
We need to tee S3 events for the Product Delivery buckets - which means events could also be delivered via SNS. We want to keep support for "native S3" events, as those are useful in other contexts.

To Do:
 - [x] add test cases
 - [x] detect source of event
 - [x] add code to extract S3 record from SNS if need be
 - [x] add some additional logging so we know which path received the event.

**Comment by hwine** [comment]
Sample of S3 event wrapped in SNS event:
```json
# Pastebin 6uLDThy0
{
   "Records" : [
      {
         "Sns" : {
            "MessageAttributes" : {},
            "UnsubscribeUrl" : "https://sns.us-east-1.amazonaws.com/?Action=Unsubscribe&SubscriptionArn=arn:aws:sns:us-east-1:142069644989:productdelivery-stage-s3events:122a172c-3e3b-496d-8e47-e0a2ed4f38b1",
            "Signature" : "Dw6ZXWI6wR2HfrADV/V4ODKS+I70nCSpc8T3YO72NUFt/xlPbJVzb2ydMi//5INIdafJcTW0bXaaD0BmeiMZ2AmowpfnXJtBXIphLz58EsW1KN7fB7ip/E5yYaguS79u3YTLYqFhi/WcdpnDWg39EywW1sN8+VxaJQ/7L/C0cvkoJPgbWVmUHZX9vqiqM2OkePLH5ji2h1k+3NxKlMaoQRZs4Z4mpMI8YfkBDmn4Ma3ZrfKgERm/sj0qjD0pFKreXcTm1Y2RFkXhAE0DP8s4A6O9+Mxmddt5KVdgIq0uZ/1hjglOht2NDXqru5rO9N2aczl8f95OWp8z1zodlTvq8Q==",
            "Message" : "{\"Records\":[{\"eventVersion\":\"2.0\",\"eventSource\":\"aws:s3\",\"awsRegion\":\"us-east-1\",\"eventTime\":\"2017-10-16T19:27:22.254Z\",\"eventName\":\"ObjectCreated:Put\",\"userIdentity\":{\"principalId\":\"AWS:AIDAIULXF63K2H2FTTMD6\"},\"requestParameters\":{\"sourceIPAddress\":\"54.214.230.165\"},\"responseElements\":{\"x-amz-request-id\":\"1D2FDA789DA2E260\",\"x-amz-id-2\":\"U1NUbqo5UTwf+Pkj+BTwc3Xuvw/mHpmVYhIWvgikM1jyYvOju8DB+nxwpiduu9yPFXVQxXuIE4o=\"},\"s3\":{\"s3SchemaVersion\":\"1.0\",\"configurationId\":\"Firefox Asset Upload\",\"bucket\":{\"name\":\"net-mozaws-stage-delivery-firefox\",\"ownerIdentity\":{\"principalId\":\"A53F4QSBA8O73\"},\"arn\":\"arn:aws:s3:::net-mozaws-stage-delivery-firefox\"},\"object\":{\"key\":\"pub/firefox/candidates/5...beetmover-checksums/update/mac/or/firefox-56.0b2-58.0b1.partial.mar.asc.beet\",\"size\":319,\"eTag\":\"704c98757a8201d30cd3aef75f811ee7\",\"versionId\":\"KXoPKO9f6A8pCs0_OHB.9aff.uVYU_Y0\",\"sequencer\":\"0059E5081A2100FEEC\"}}}]}",
            "Subject" : "Amazon S3 Notification",
            "Type" : "Notification",
            "SigningCertUrl" : "https://sns.us-east-1.amazonaws.com/SimpleNotificationService-433026a4050d206028891664da859041.pem",
            "Timestamp" : "2017-10-16T19:27:22.316Z",
            "TopicArn" : "arn:aws:sns:us-east-1:142069644989:productdelivery-stage-s3events",
            "SignatureVersion" : "1",
            "MessageId" : "349f1147-d0b6-5367-a9a1-4318daec8389"
         },
         "EventVersion" : "1.0",
         "EventSubscriptionArn" : "arn:aws:sns:us-east-1...",
         "EventSource" : "aws:sns"
      }
   ]
}
```

**Comment by hwine** [comment]
Because this code will execute outside the context of an object, decided to log into the response framework (slightly hacky). If an SNS event is processed, the following response will have the following line added:
```
  "unpacker": "SNS message unpacked",
```

**Comment by hwine** [comment]
fixed by 70aaf7e13f437636b479155a7301c1e9bfbeae14
